Python/compile.c

```diff
@@ -7131,20 +7188,21 @@ stackdepth(basicblock *entryblock, int code_flags)
         basicblock *next = b->b_next;
         for (int i = 0; i < b->b_iused; i++) {
             struct instr *instr = &b->b_instr[i];
-            int effect = stack_effect(instr->i_opcode, instr->i_oparg, 0);
+            int oparg = OPARG_VALUE(instr->i_oparg);
+            int effect = stack_effect(instr->i_opcode, oparg, 0);
             if (effect == PY_INVALID_STACK_EFFECT) {
                 PyErr_Format(PyExc_SystemError,
                              "compiler stack_effect(opcode=%d, arg=%i) failed",
-                             instr->i_opcode, instr->i_oparg);
+                             instr->i_opcode, oparg);
                 return -1;
             }
             int new_depth = depth + effect;
             assert(new_depth >= 0); /* invalid code or bug in stackdepth() */
             if (new_depth > maxdepth) {
                 maxdepth = new_depth;
             }
-            if (HAS_TARGET(instr->i_opcode)) {
-                effect = stack_effect(instr->i_opcode, instr->i_oparg, 1);
+            if (instr->i_oparg.type == JUMP_TARGET_ARG) {
```

Comment:
We don't feel the need to assert `HAS_TARGET` here?

Of course if the `OPARG_VALUE` above had already done an assertion of compatible arg types, it would be redundant.